great_expectations/checkpoint/util.py

```diff
@@ -239,76 +272,247 @@ def get_runtime_batch_request(
-    }
-    if validation_dict.get("name") is not None:
-        substituted_validation_dict["name"] = validation_dict["name"]
-    validate_validation_dict(substituted_validation_dict)
-    return substituted_validation_dict
+        dest_config["batch_request"] = updated_batch_request
+    if source_config.get("action_list") is not None:
+        action_list = dest_config.get("action_list") or []
+        dest_config["action_list"] = get_updated_action_list(
+            base_action_list=action_list,
+            other_action_list=source_config["action_list"],
+        )
+    if source_config.get("evaluation_parameters") is not None:
+        evaluation_parameters = dest_config.get("evaluation_parameters") or {}
+        updated_evaluation_parameters = nested_update(
+            evaluation_parameters,
+            source_config["evaluation_parameters"],
+        )
+        dest_config["evaluation_parameters"] = updated_evaluation_parameters
+    if source_config.get("runtime_configuration") is not None:
+        runtime_configuration = dest_config.get("runtime_configuration") or {}
+        updated_runtime_configuration = nested_update(
+            runtime_configuration,
+            source_config["runtime_configuration"],
+        )
+        dest_config["runtime_configuration"] = updated_runtime_configuration
+    if source_config.get("validations") is not None:
+        validations = dest_config.get("validations") or []
+        existing_validations = template_config.get("validations") or []
+        validations.extend(
+            filter(
+                lambda v: v not in existing_validations, source_config["validations"]
+            )
+        )
+        dest_config["validations"] = validations
+    if source_config.get("profilers") is not None:
+        profilers = dest_config.get("profilers") or []
+        existing_profilers = template_config.get("profilers") or []
+        profilers.extend(
+            filter(lambda v: v not in existing_profilers, source_config["profilers"])
+        )
+        dest_config["profilers"] = profilers
+
+    return dest_config
+
+
+def substitute_runtime_config(source_config: dict, runtime_kwargs: dict) -> dict:
+    if not (runtime_kwargs and any(runtime_kwargs.values())):
+        return source_config
+
+    dest_config: dict = copy.deepcopy(source_config)
+
+    # replace
+    if runtime_kwargs.get("template_name") is not None:
+        dest_config["template_name"] = runtime_kwargs["template_name"]
+    if runtime_kwargs.get("run_name_template") is not None:
+        dest_config["run_name_template"] = runtime_kwargs["run_name_template"]
+    if runtime_kwargs.get("expectation_suite_name") is not None:
+        dest_config["expectation_suite_name"] = runtime_kwargs["expectation_suite_name"]
+    if runtime_kwargs.get("expectation_suite_ge_cloud_id") is not None:
+        dest_config["expectation_suite_ge_cloud_id"] = runtime_kwargs[
+            "expectation_suite_ge_cloud_id"
+        ]
```

Comment:
Same pattern here and below

Reply:
@cdkini Could you please put this idea into a separate JIRA ticket for us to follow up on?  Thanks!